### DISCUSSION TITLE
Execute a common code during start of any dag execution or during clearing any tasks for dag rerun

### DISCUSSION THREAD

**Discussion by chnaveen138** [apache/airflow#35866] [Q&A]
Hi,

I have a requirement where I need to execute some code during start of the dags execution. This same code is applicable for all the dags.

In my case I need to treat even a task clearance in already completed dag execution also as a new dag run and it should execute that common code before starting. So having a common start task across all my dags and executing this common code as part of it will not work in my case since we cannot always expect developer to clear start task during rerun or clearing of the dag execution. I couldn't find a good solution for this.

Basically in my case, why I want this functionality is I need to restrict new dag run executions if a same dag run execution is already in_progress for that logical_date. This can happen usually during manual runs.

If there is any callback functionality like `on_start_callback` similar to `on_success_callback` in airflow would've fixed my problem.

Any help would be appreciated. Kind of stuck with this problem for few days.

**Comment by potiuk** [comment]
https://airflow.apache.org/docs/apache-airflow/stable/administration-and-deployment/listeners.html#listeners
